File: Quokka.RISCV.CS2CPP.Tests/Make/SOCBlinker.cpp
#include "SOCBlinker.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace SOCBlinker
{
	void Firmware::EntryPoint()
	{
		while(true)
		{
			unsigned int counter = SOCBlinker_SOC_Counter;
			(counter++);
			SOCBlinker_SOC_Counter = counter;
		}
	}

	unsigned int Firmware::result = 0;
	unsigned int Firmware::buff[10] = {0};
}

File: Quokka.RISCV.CS2CPP.Tests/Source/SOCBlinker.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SOCBlinker
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual byte Blinker { get; set; }
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            uint counter = 0;
            while(true)
            {
                counter++;
                SOC.Instance.Blinker = (byte)(counter >> 24);
            }
        }
    }
}
